src/commons_codec/transform/mongodb.py

```diff
@@ -0,0 +1,203 @@
+        # https://www.mongodb.com/docs/manual/changeStreams/#lookup-full-document-for-update-operations
+        elif operation_type in ["update", "replace"]:
+            full_document = self.get_full_document(record)
+            values_clause = self.full_document_to_values(full_document)
+            where_clause = self.where_clause(record)
+            sql = f"UPDATE {self.table_name} SET {self.DATA_COLUMN} = '{values_clause}' WHERE {where_clause};"
+
+        elif operation_type == "delete":
+            where_clause = self.where_clause(record)
+            sql = f"DELETE FROM {self.table_name} WHERE {where_clause};"
+
+        # TODO: Enable applying the "drop" operation conditionally when enabled.
+        elif operation_type == "drop":
+            logger.info("Received 'drop' operation, but skipping to apply 'DROP TABLE'")
+            sql = ""
+
+        elif operation_type == "invalidate":
+            logger.info("Ignoring 'invalidate' CDC operation")
+            sql = ""
+
+        else:
+            raise ValueError(f"Unknown CDC operation type: {operation_type}")
+
+        return sql
+
+    @staticmethod
+    def get_document_key(record: t.Dict[str, t.Any]) -> str:
+        """
+        Return value of document key (MongoDB document OID) from CDC record.
+
+        "documentKey": {"_id": ObjectId("669683c2b0750b2c84893f3e")}
+        """
+        return str(record.get("documentKey", {}).get("_id"))
+
+    @staticmethod
+    def get_full_document(record: t.Dict[str, t.Any]) -> t.Dict[str, t.Any]:
+        """
+        return `fullDocument` representation from record.
+        """
+        return t.cast(dict, record.get("fullDocument"))
+
+    def full_document_to_values(self, document: t.Dict[str, t.Any]) -> str:
+        """
+        Serialize CDC event's "fullDocument" representation to a `VALUES` clause in CrateDB SQL syntax.
+
+        IN (top-level stripped):
+        "fullDocument": {
+            "_id": ObjectId("669683c2b0750b2c84893f3e"),
+            "id": "5F9E",
+            "data": {"temperature": 42.42, "humidity": 84.84},
+            "meta": {"timestamp": datetime.datetime(2024, 7, 11, 23, 17, 42), "device": "foo"},
+        }
+
+        OUT:
+        {"_id": {"$oid": "669683c2b0750b2c84893f3e"},
+         "id": "5F9E",
+         "data": {"temperature": 42.42, "humidity": 84.84},
+         "meta": {"timestamp": {"$date": "2024-07-11T23:17:42Z"}, "device": "foo"},
+        }
+        """
+        return json.dumps(self.deserialize_item(document))
+
+    def where_clause(self, record: t.Dict[str, t.Any]) -> str:
+        """
+        When converging an oplog of a MongoDB collection, the primary key is always the MongoDB document OID.
+
+        IN (top-level stripped):
+        "documentKey": {"_id": ObjectId("669683c2b0750b2c84893f3e")}
+
+        OUT:
+        WHERE oid = '669683c2b0750b2c84893f3e'
+        """
+        oid = self.get_document_key(record)
+        return f"oid = '{oid}'"
```

Comment:
Dear @hlcianfagna, @hammerhead, @surister, and @wierdvanderhaar,

may I kindly ask you to review those routines? If you worked with DynamoDB CDC convergence already, they might feel familiar to you.

Thanks in advance,
Andreas.

Reply:
### DDL schema
In particular, I am interested if you agree with the SQL DDL schema and the corresponding update strategy, or if you would prefer a different one, and how that would look like.
```sql
CREATE TABLE <tablename> (oid TEXT, data OBJECT(DYNAMIC));
```

### Value serialization
Currently, typed values are retained in their nested-JSON forms, so a sample content of the `data` column looks like outlined below. Here, I would be interested what you think about the nestings of `"timestamp": {"$date": "2024-07-11T23:17:42Z"}` and friends, how MongoDB's special values are serialized like.

- Is it good to go for a first iteration?
- Should there be a subsequent iteration to address any shortcomings you may report?

```json
{
  "_id": {
    "$oid": "6696c937f6e1770586480277"
  },
  "data": {
    "humidity": 84.84,
    "temperature": 42.42
  },
  "id": "5F9E",
  "meta": {
    "device": "foo",
    "timestamp": {
      "$date": "2024-07-11T23:17:42Z"
    }
  }
}
```